sphinx/config.py

```diff
@@ -257,6 +257,7 @@ class Config(object):
         linkcheck_timeout = (None, None, [int]),
         linkcheck_workers = (5, None),
         linkcheck_anchors = (True, None),
+        linkcheck_anchors_ignore = ([], None),
```

Comment:
Could we make this default to `['!']`, so we can delete this hard-coded path from the code?